### PR TITLE
[54-74] `LazySinkSpec`

### PR DESCRIPTION
**Pull request by eaba**
## Changes

- Change `AssertAllStagesStopped` to `AssertAllStagesStoppedAsync`

### CODE REVIEW

File: src/core/Akka.Streams.Tests/Dsl/LazySinkSpec.cs [hunk 1]
```diff
@@ -52,7 +52,12 @@ public async Task A_LazySink_must_work_in_the_happy_case()
                 var taskProbe = Source.From(Enumerable.Range(0, 11)).RunWith(lazySink, Materializer);
                 var probe = await taskProbe.ShouldCompleteWithin(RemainingOrDefault);
                 probe.Value.Request(100);
-                Enumerable.Range(0, 11).ForEach(i => probe.Value.ExpectNext(i));
+                foreach (var i in Enumerable.Range(0, 11)) 
+                {
+                    await probe.Value.ExpectNextAsync(i);
+                }
+                //I CHECK IT THIS IS NOT GOOD - EVERYTHING WORKS FINE NOW
```
Comment (eaba) [inline src/core/Akka.Streams.Tests/Dsl/LazySinkSpec.cs]: //I CHECK IT THIS IS NOT GOOD - EVERYTHING WORKS FINE NOW

File: src/core/Akka.Streams.Tests/Dsl/LazySinkSpec.cs [hunk 2]
```diff
@@ -44,53 +44,67 @@ private static Func<TMat> Fallback<TMat>()
...
-                probe.ExpectNext(0);
-                Enumerable.Range(1,10).ForEach(i =>
+                await probe.ExpectNextAsync(0);
+
+                foreach (var i in Enumerable.Range(1, 10))
+                {
+                    sourceSub.SendNext(i);
+                    await probe.ExpectNextAsync(i);
+                }
+                //I CHECK IT THIS IS NOT GOOD - EVERYTHING WORKS FINE NOW
```
Comment (Aaronontheweb) [inline src/core/Akka.Streams.Tests/Dsl/LazySinkSpec.cs]: Please remove any comments that aren't functional

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Needs some cleanup but otherwise looks good

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**